### PR TITLE
Cleaned up code and added filtering

### PR DESCRIPTION
**Pull request by jordyob03**
Added filtering : https://app.asana.com/0/1205465127204334/1205958709776184
Fixed Messages: https://app.asana.com/0/1205465127204334/1205907689503697

### CODE REVIEW

File: include/waypointSenderFunctionsLib.h [hunk 1]
```diff
@@ -15,8 +15,8 @@
 
 geometry_msgs::Point enu_2_local(nav_msgs::Odometry current_pose_enu);
 geometry_msgs::Point get_current_location(nav_msgs::Odometry current_pose);
-geometry_msgs::PoseStamped set_heading(float heading, float local_desired_heading, float correction_heading, geometry_msgs::PoseStamped waypoint);
-geometry_msgs::PoseStamped set_destination(float x, float y, float z, float psi, float correction_heading, float local_desired_heading, geometry_msgs::Pose correction_vector, geometry_msgs::Point local_offset_pose, geometry_msgs::PoseStamped waypoint);
+geometry_msgs::PoseStamped set_heading(float headingg, geometry_msgs::PoseStamped waypoint);
```
Comment (gpearcey) [inline include/waypointSenderFunctionsLib.h]: is headingg with 2 g's a typo?

File: lib/waypointSenderFunctionsLib.cpp [hunk 2]
```diff
@@ -69,23 +69,17 @@ geometry_msgs::PoseStamped set_heading(float heading, float local_desired_headin
...
-	x = Xlocal + correction_vector.position.x;
-	y = Ylocal + correction_vector.position.y;
-	z = Zlocal + correction_vector.position.z;
+	float Xlocal = x*cos((90)*deg2rad) - y*sin((90)*deg2rad);
+	float Ylocal = x*sin((90)*deg2rad) + y*cos((90)*deg2rad);
 
-	waypoint.pose.position.x = -x;
-	waypoint.pose.position.y = -y;
-	waypoint.pose.position.z = z;
+	waypoint.pose.position.x = Xlocal;
```
Comment (gpearcey) [inline lib/waypointSenderFunctionsLib.cpp:80]: Why are the negatives removed here? Were they even supposed to be there originally I just forget.

File: src/wp_filter_sender.cpp [hunk 3]
```diff
@@ -190,56 +147,46 @@ int arm()
...
 
 	ROS_INFO("Starting mission");
+
 	while(ros::ok())
 	{
 		ros::spinOnce();
 		rate.sleep();
 
-		desired_waypoint = set_destination(goal_pos.x, goal_pos.y, goal_pos.z, goal_pos.psi, correction_heading, local_desired_heading, correction_vector, local_offset_pose, desired_waypoint);
+		desired_waypoint = set_destination(goal_pos.x, goal_pos.y, goal_pos.psi, desired_waypoint);
```
Comment (gpearcey) [inline src/wp_filter_sender.cpp:189]: Is there any checking to make sure the waypoint comes from the correct task?

File: lib/waypointSenderFunctionsLib.cpp (hunk 2 above)
Comment (jordyob03) [reply lib/waypointSenderFunctionsLib.cpp:80]: Sorry meant to mention this to you, I figured out why we needed to add these negatives. There was a mistake in what the original writer of this code had. If you look above I changed the -90 in the formula to 90 and that fixed the issue. Basically the waypoints need to be rotated 90 degrees to match Gazebo but they were being rotated the wrong way.

**Review by gpearcey — APPROVED** [review]

**Pull request merged**